tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -281,60 +278,112 @@ private void SetRemoteConfigCapabilites()
-                e.Acknowledge(asmDataConfig.Name);
-            }
+                _asmDataConfigs ??= new Dictionary<string, RcmModels.AsmData.Payload>();
+                var asmDataConfigs = e.GetDeserializedConfigurations<RcmModels.AsmData.Payload>();
+                foreach (var asmDataConfig in asmDataConfigs)
+                {
+                    _asmDataConfigs[asmDataConfig.Name] = asmDataConfig.TypedFile;
+                    e.Acknowledge(asmDataConfig.Name);
+                }
 
-            var updated = UpdateRulesData();
-            foreach (var asmDataConfig in asmDataConfigs)
-            {
-                if (!updated)
+                var updated = UpdateRulesData();
+                foreach (var asmDataConfig in asmDataConfigs)
                 {
-                    e.Error(asmDataConfig.Name, "Waf could not update the rules");
+                    if (!updated)
+                    {
+                        e.Error(asmDataConfig.Name, "Waf could not update the rules");
+                    }
+                    else
+                    {
+                        e.Acknowledge(asmDataConfig.Name);
+                    }
                 }
-                else
+            }
+        }
+
+        private void AsmProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
+        {
+            if (!_enabled) { return; }
+
+            lock (_settings)
+            {
+                _ruleStatus ??= new Dictionary<string, bool>(StringComparer.InvariantCultureIgnoreCase);
+                var asmConfigs = e.GetDeserializedConfigurations<RcmModels.Asm.Payload>();
+                int ruleCount = 0;
+                foreach (var asmConfig in asmConfigs)
                 {
-                    e.Acknowledge(asmDataConfig.Name);
+                    try
+                    {
+                        var rulesStatus = asmConfig.TypedFile.RuleStatus;
+                        foreach (var data in rulesStatus)
+                        {
+                            if (data.Id == null || data.Enabled == null)
+                            {
+                                continue;
+                            }
+
+                            _ruleStatus[data.Id] = data.Enabled.Value;
+                            ruleCount++;
+                        }
+
+                        if (ruleCount > 0)
+                        {
+                            e.Acknowledge(asmConfig.Name);
+                        }
+                        else
+                        {
+                            e.Error(asmConfig.Name, "No valid Waf rule status data received.");
+                        }
+                    }
+                    catch (Exception err)
+                    {
+                        e.Error(asmConfig.Name, "Waf rule status data error: " + err.Message);
+                    }
                 }
+
+                _waf.ToggleRules(_ruleStatus);
```

Comment:
could be interesting to report the failure to update the WAF as an error.

Reply:
I agree

Reply:
"Fixed" in 81aae49427c20400fb57cd3cca38d48039d39efe